pkg/bearertoken/grpc.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package bearertoken
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const Key = "bearer.token"
+
+type tokenatedServerStream struct {
+	grpc.ServerStream
+	context context.Context
```

Comment:
this is never used

Reply:
The `tokenatedServerStream `struct is actually used in the `streamServerInterceptor `function, where it's needed to wrap the original `ServerStream `with a context that includes the bearer token.

Reply:
The context field of this struct is never used, so the whole struct is unnecessary. Injecting token into context is done by manipulating the context, not by capturing it.

Reply:
per slack discussion, you need to implement the `Context()` method for your struct, only then you will be able to expose the new `context` to the downstream handler.